src/Microsoft.CodeAnalysis.Analyzers/Core/MetaAnalyzers/SymbolIsBannedInAnalyzersAnalyzer.cs

```diff
@@ -0,0 +1,123 @@
+            compilationContext.RegisterSymbolAction(analyzePossibleAnalyzerOrGenerator, SymbolKind.NamedType);
+
+            const string fileName = "Microsoft.CodeAnalysis.Analyzers.AnalyzerBannedSymbols.txt";
+            var stream = typeof(SymbolIsBannedInAnalyzersAnalyzer<>).Assembly.GetManifestResourceStream(fileName);
+            var source = SourceText.From(stream);
+            var result = new Dictionary<ISymbol, BanFileEntry>();
+            foreach (var line in source.Lines)
+            {
+                var text = line.ToString();
+                if (string.IsNullOrWhiteSpace(text))
+                {
+                    continue;
+                }
+
+                var entry = new BanFileEntry(text, line.Span, source, fileName);
+                var symbols = DocumentationCommentId.GetSymbolsForDeclarationId(entry.DeclarationId, compilationContext.Compilation);
+                if (!symbols.IsDefaultOrEmpty)
+                {
+                    foreach (var symbol in symbols)
+                    {
+                        result.Add(symbol, entry);
+                    }
+                }
+            }
+
+            return result;
+
+            void analyzePossibleAnalyzerOrGenerator(SymbolAnalysisContext symbolAnalysisContext)
+            {
+                var symbol = symbolAnalysisContext.Symbol;
+
+                var attributes = symbol.GetAttributes();
+                if (attributes.Any(shouldReportNotSpecifiedEnforceAnalyzerBannedApisSetting))
+                {
+                    symbolAnalysisContext.ReportDiagnostic(symbol.Locations.CreateDiagnostic(SymbolIsBannedInAnalyzersAnalyzer.NoSettingSpecifiedSymbolIsBannedRule, symbol));
+                }
+
+                bool shouldReportNotSpecifiedEnforceAnalyzerBannedApisSetting(AttributeData attributeData)
+                {
+                    var treeOptions = symbolAnalysisContext.Options.AnalyzerConfigOptionsProvider.GetOptions(attributeData.ApplicationSyntaxReference.SyntaxTree);
+                    var categorizedTreeOptions = SyntaxTreeCategorizedAnalyzerConfigOptions.Create(treeOptions);
+                    var enforceBannedApisIsSpecified = categorizedTreeOptions.TryGetOptionValue(
+                        EditorConfigOptionNames.EnforceExtendedAnalyzerRules,
+                        OptionKind.DotnetCodeQuality,
+                        rule: null,
+                        tryParseValue: bool.TryParse,
+                        false,
+                        out _);
+                    if (enforceBannedApisIsSpecified)
+                    {
+                        return false;
+                    }
+
+                    return attributeData.AttributeClass.Equals(diagnosticAnalyzerAttributeType, SymbolEqualityComparer.Default)
+                        || attributeData.AttributeClass.Equals(generatorAttributeType, SymbolEqualityComparer.Default);
```

Comment:
Should we move this attribute type check upfront in this method? I presume it will be cheaper then the code to look for the option setting?